src/deepsparse/loggers/metric_functions/helpers/config_generation.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Helper functions for generating metric function configs
+"""
+
+__all__ = ["data_logging_config_from_predefined"]
+
+import logging
+import os
+import textwrap
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+
+from deepsparse.loggers.build_logger import predefined_metric_function
+from deepsparse.loggers.config import MetricFunctionConfig
+from deepsparse.loggers.metric_functions.registry import DATA_LOGGING_REGISTRY
+
+
+_WHITESPACE = "  "
+_LOGGER = logging.getLogger(__name__)
+
+
+def data_logging_config_from_predefined(
```

Comment:
Was having pre-built yaml files that we edit by hand discussed as an alternative to this approach? It'd be a lot easier to edit a single yaml file than tweak these functions and have to double check that none of the other auto-generated yamls have changed. Especially if each of the yamls has slight differences in the structure that are hard to capture in this kind of serialization code

Reply:
they were, ironically went with auto for the purpose of easier maintenance. And the hope was for the auto generated configs to be used as examples. 

@corey-nm (or @dbogunowicz ) do we expect these functions to have differences in structure in a way that the auto generated wouldn't capture? most functions should just be a name, target pipeline, and target identifier, no?

Do you see a difference between needing to manually edit manual files vs auto generating when an expected change occurs?

Additionally we can add tests to compare fresh auto generated configs vs the example docs that way GHA can catch and remind us to auto generate when they become stale or, if an unintended change was made that affects the config, a failed test would flag this as well (@dbogunowicz thoughts on this test)

Reply:
Let's sync on that offline.